### ISSUE TITLE
Automatic review catalog reaping

### ISSUE THREAD

**Issue by karlcz** [nih-cfde/cfde-deriva#111]
Assuming the ingest workflow ends with a submission loaded into registry and built as review catalog (if not a pre-catalog failure), define and add mechanisms to handle cleanup of review catalogs based on lifecycle status

- Purge soon after some terminal failure status is reached?
- Allow a failed-hold status to keep (partial) results for possible problem determination, with user input releasing this hold state?
- Have timeout based reaping even for held failures?
- Have a cleanup policy for released content?
- What about rejected catalogs (not ingest failures)?

It seems plausible we wouldl want to keep a successful review catalog not only trough the review period but even after approval, for as long as it is still pending release. Once the same content has been released, the review catalog is certainly redundant.  For rejected submissions, should we delete quickly or have some hold period before we clear them?  Or, does it depend on whether a subsequent submission has been made...?

**Comment by karlcz** [comment]
In prior discussions, I understood most of this to be declalred out of scope for epic 2. Right @lliming ?  This, I think I should remove the epic 2 association from this issue and just leave it in backlog for future work.

**Comment by lliming** [comment]
Yes, this doesn't need to happen for Epic 2, so it can be left on the backlog.

**Comment by karlcz** [comment]
We have a prototype expiration script in app-dev which successfully does some of these things via SQL set operations...
1. Start with a set of candidates to purge
2. Except, disqualify a set of candidates matching other criteria
3. Update the CFDE registry metadata with the planned purge
4. Update the ermrest internal registry to emulate a DELETE /ermrest/catalong/N call for each purged catalog
5. Wait for the usual ermrest maintenance to actually prune/drop the victims

The initial set of candidates is a UNION of these categories:
- Submissions obsoleted by a newer "approved" submission for the same DCC
- Submissions in a failed state (some may have an associated catalog)
- Submissions older than 1 month from time the script runs

The set of disqualified candidates is a UNION of these categories:
- Constituents of a pending (content-ready) release
- Submissions submitted in last 2 weeks
- Most recent submission for each DCC, regardless of status
- Most recent failed submission for each DCC
